### PR TITLE
Start RMC services before migration

### PR DESCRIPTION
**Pull request by abdhaleegit**
RMC service is must for migration to happen, make sure it gets
activated while or in between the test

Signed-off-by: Abdul Haleem <abdhalee@linux.vnet.ibm.com>

### CODE REVIEW

File: io/net/virt-net/lpm.py [hunk 1]
```diff
@@ -66,6 +66,7 @@ def setUp(self):
         '''
         self.install_packages()
         self.rsct_service_start()
+        self.rmc_service_start()
```
Comment (narasimhan-v) [inline io/net/virt-net/lpm.py]: @abdhaleegit My thoughts: This is a workaround, so we should not be PASSing the test in this scenario.
We can check if the rsct services are active, and if inactive, set a flag. Restart only if necessary
At the end, we can check if flag is set, WARN/FAIL the test based on that. This also makes sure that LPM continues and failure is logged.

Let me know if this suggestion is fine with you.

**Comment by narasimhan-v** [comment]
@abdhaleegit Is this still being pursued ?

**Comment by narasimhan-v** [comment]
@abdhaleegit updates please ?\

**Comment by abdhaleegit** [comment]
Sorry for the delay, will be looking into this

File: io/net/virt-net/lpm.py (hunk 1 above)
Comment (abdhaleegit) [reply io/net/virt-net/lpm.py:69]: RMC service going down after a migration is a bug and I have raised it against HMC,   

for the test there will be a check for rmc after every migration and WARN test if rmc is inactive and also make sure the rmc is restarted for other test to continue (this makes sure the migration is complete even after warn)

@narasimhan-v

Comment (abdhaleegit) [reply io/net/virt-net/lpm.py]: With the latest code, I have taken care of above suggestion

**Comment by abdhaleegit** [comment]
@narasimhan-v Please have a look at the new approach

File: io/net/virt-net/lpm.py [hunk 2]
```diff
@@ -56,7 +58,9 @@ def setUp(self):
         self.lpar = self.get_partition_name("Partition Name")
         if not self.lpar:
             self.cancel("LPAR Name not got from lparstat command")
-
+        self.lpar_ip = self.get_mcp_component("MNName")
+        if not self.lpar_ip:
+            self.cancel("LPAR IP not got from lsrsrc command")
```
Comment (narasimhan-v) [inline io/net/virt-net/lpm.py:63]: Is this needed, since we already got lpar name from lparstat command

**Review by narasimhan-v — CHANGES_REQUESTED** [review]
@abdhaleegit This approach seems fine. Just a minor clarification for lpar name.

File: io/net/virt-net/lpm.py (hunk 2 above)
Comment (abdhaleegit) [reply io/net/virt-net/lpm.py:63]: @narasimhan-v the diagrmc needs lpar ip address to check for RMC connection, so we need both name and ip

Comment (narasimhan-v) [reply io/net/virt-net/lpm.py:63]: Ah okay.

**Comment by narasimhan-v** [comment]
@abdhaleegit Sure, once log is updated, this is ready to be merged.

**Comment by abdhaleegit** [comment]
'''
[root@ltc-zz3-lp1 virt-net]# avocado run lpm.py --mux-yaml lpm.py.data/lpm.yaml 
JOB ID     : 2584a7063200321e032a865467c19cb932fb51e7
JOB LOG    : /home/ioci/avocado-fvt-wrapper/results/job-2020-09-24T10.04-2584a70/job.log
 (1/2) lpm.py:LPM.test_migrate;run-e74c: PASS (499.84 s)
 (2/2) lpm.py:LPM.test_migrate_back;run-e74c: PASS (504.08 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/ioci/avocado-fvt-wrapper/results/job-2020-09-24T10.04-2584a70/results.html
JOB TIME   : 1004.74 s

**Comment by abdhaleegit** [comment]
[lpmfixlogs.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/5277212/lpmfixlogs.txt)

**Comment by abdhaleegit** [comment]
```
11:01:38 INFO    : Running Host Tests Suite io_vnic_lpm_fvt_lpm_lpm_5_iterations
11:01:38 INFO    : Running: /usr/local/bin/avocado run avocado-misc-tests/io/net/virt-net/lpm.py -m /tmp/mux/io_vnic_lpm_fvt_lpm_lpm_5_iterations.yaml --force-job-id 719c485b9c35f353cd11a3198315e6dad41952c8  --job-results-dir /home/ioci/logs/io_vnic_lpm_fvt/results --execution-order tests-per-variant
JOB ID     : 719c485b9c35f353cd11a3198315e6dad41952c8
JOB LOG    : /home/ioci/logs/io_vnic_lpm_fvt/results/job-2020-09-24T11.01-719c485/job.log
 (01/10) avocado-misc-tests/io/net/virt-net/lpm.py:LPM.test_migrate;run-iteration-default-1-6c80:  PASS (497.24 s)
 (02/10) avocado-misc-tests/io/net/virt-net/lpm.py:LPM.test_migrate_back;run-iteration-default-1-6c80:  PASS (510.08 s)
 (03/10) avocado-misc-tests/io/net/virt-net/lpm.py:LPM.test_migrate;run-iteration-default-2-280c:  PASS (512.50 s)
 (04/10) avocado-misc-tests/io/net/virt-net/lpm.py:LPM.test_migrate_back;run-iteration-default-2-280c:  PASS (505.08 s)
 (05/10) avocado-misc-tests/io/net/virt-net/lpm.py:LPM.test_migrate;run-iteration-default-3-763e:  PASS (496.99 s)
 (06/10) avocado-misc-tests/io/net/virt-net/lpm.py:LPM.test_migrate_back;run-iteration-default-3-763e:  PASS (518.31 s)
 (07/10) avocado-misc-tests/io/net/virt-net/lpm.py:LPM.test_migrate;run-iteration-default-4-9540:  PASS (498.64 s)
 (08/10) avocado-misc-tests/io/net/virt-net/lpm.py:LPM.test_migrate_back;run-iteration-default-4-9540:  PASS (525.22 s)
 (09/10) avocado-misc-tests/io/net/virt-net/lpm.py:LPM.test_migrate;run-iteration-default-5-5ff5:  PASS (496.20 s)
 (10/10) avocado-misc-tests/io/net/virt-net/lpm.py:LPM.test_migrate_back;run-iteration-default-5-5ff5:  PASS (504.66 s)
RESULTS    : PASS 10 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/ioci/logs/io_vnic_lpm_fvt/results/job-2020-09-24T11.01-719c485/results.html
JOB TIME   : 5065.96 s

**Review by narasimhan-v — APPROVED** [review]

**Pull request merged**